src/python/pants/engine/internals/native_engine.pyi

```diff
@@ -110,7 +110,7 @@ def capture_snapshots(
     scheduler: PyScheduler,
     session: PySession,
     path_globs_and_root_tuple_wrapper: _PathGlobsAndRootCollection,
-) -> tuple[PySnapshot, ...]: ...
+) -> list[PySnapshot]: ...
```

Comment:
Given that PyO3 returns `Vec<>`, I figured we can lean into it - not worth the eager type conversion to tuple.